backend/core/src/primary-adapters/rest/video/video-controller.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  BaseHttpController,
+  controller,
+  httpGet,
+  httpPost,
+  request,
+  requestBody,
+  requestParam,
+} from 'inversify-express-utils';
+import { inject } from 'inversify';
+import {
+  CONTAINER_TYPES,
+  CreateReactionRequestDto,
+  CreateReactionResponseDto,
+  ExtendedAuthenticatedRequest,
+  VideoCommentResponseDto,
+} from '~/shared/types/types';
+import { HttpError, VideoApiPath, VideoBaseResponseDto, VideoCommentRequestDto } from 'shared/build';
+import { VideoService } from '~/core/video/aplication/video-service';
+import { Video } from '@prisma/client';
+import { authenticationMiddleware, optionalAuthenticationMiddleware } from '~/primary-adapters/rest/middleware';
+
+/**
+ * @swagger
+ * tags:
+ *   name: video
+ *   description: video
+ * components:
+ *    schemas:
+ *      CreateReactionRequestDto:
+ *        type: object
+ *        properties:
+ *          isLike:
+ *            type: boolean
+ *        required:
+ *          - isLike
+ *      VideoCommentRequestDto:
+ *        type: object
+ *        properties:
+ *          videoId:
+ *            type: string
+ *          text:
+ *            type: string
+ *        required:
+ *          - videoId
+ *          - text
+ */
+
+@controller('/videos')
+export class VideoController extends BaseHttpController {
+  private videoService: VideoService;
+
+  constructor(@inject(CONTAINER_TYPES.VideoService) videoService: VideoService) {
+    super();
+
+    this.videoService = videoService;
+  }
+
+  /**
+   * @swagger
+   * /videos/{id}:
+   *    get:
+   *      tags:
+   *        - video
+   *      operationId: getVideoById
+   *      security:
+   *      - bearerAuth: []
+   *      consumes:
+   *        - application/json
+   *      produces:
+   *        - application/json
+   *      description: Get video data by video id
+   *      parameters:
+   *        - in: path
+   *          name: id
+   *          description: video ID to get the data of this video
+   *          required: true
+   *          schema:
+   *            type: string
+   *      responses:
+   *        '200':
+   *          description: successful operation
+   *          content:
+   *            application/json:
+   *              schema:
+   *                $ref: '#/components/schemas/VideoBaseResponseDto'
+   *        '404':
+   *          description: video not found
+   */
+
+  @httpGet('/:id', optionalAuthenticationMiddleware)
+  public async get(
+    @requestParam('id') id: string,
+    @request() req: ExtendedAuthenticatedRequest,
+  ): Promise<VideoBaseResponseDto> {
+    let userId: string | undefined = undefined;
+    if (req.user) {
```

Comment:
If I remember correctly, `ExtendedAuthenticatedRequest` is designated for requests that are definitely authenticated, and `req.user` is not optional (unless I haven't noticed and you changed  `ExtendedAuthenticatedRequest`, but don't do it). There is `ExtendedRequest` that has user as optional.

Reply:
done